app/src/main/res/layout/dialog_for_barcode.xml

```diff
@@ -0,0 +1,60 @@
+<LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    android:layout_width="match_parent"
+    android:layout_height="wrap_content"
+    android:background="@color/white"
+    android:orientation="vertical">
+
+    <TextView
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:background="@color/white"
+        android:contentDescription="@string/app_name"
+        android:padding="@dimen/padding_normal"
+        android:scaleType="center"
+        android:textAlignment="center"
+        android:gravity="center"
+        android:text="@string/can_add_barcode"
+        android:textColor="@color/md_black_1000"
+        android:textSize="@dimen/font_large" />
+
+    <EditText
+        android:id="@+id/barcode"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:padding="@dimen/padding_short"
+       android:layout_margin="@dimen/spacing_tiny"
+        android:background="@color/white"
+        android:hint="@string/hintBarcode"
+        android:inputType="textEmailAddress" />
```

Comment:
Can you change this to be "number". We want to make sure that the user can not put in any characters that may cause the app to crash. You might want to think about checking the output of this too, to ensure that people do not find a way of pasting some unsupported characters in here